diff --git a/boost/gil/extension/io/tiff/detail/reader_backend.hpp b/boost/gil/extension/io/tiff/detail/reader_backend.hpp
index d2a308f..6b15201 100644
--- a/boost/gil/extension/io/tiff/detail/reader_backend.hpp
+++ b/boost/gil/extension/io/tiff/detail/reader_backend.hpp
@@ -104,13 +104,11 @@ public:
 
         io_error_if( _io_dev.template get_property<tiff_resolution_unit>( _info._resolution_unit) == false
           , "cannot read tiff tag");
-        io_error_if( _io_dev. template get_property<tiff_x_resolution>( _info._x_resolution ) == false
-          , "cannot read tiff tag" );
-        io_error_if( _io_dev. template get_property<tiff_y_resolution>( _info._y_resolution ) == false
-          , "cannot read tiff tag" );
 
         /// optional and non-baseline properties below here
-        _io_dev. template get_property <tiff_icc_profile> ( _info._icc_profile );
+        _io_dev. template get_property<tiff_x_resolution>( _info._x_resolution );
+        _io_dev. template get_property<tiff_y_resolution>( _info._y_resolution );
+        _io_dev. template get_property<tiff_icc_profile >( _info._icc_profile );
     }
 
     /// Check if image is large enough.
