diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f5eaec..592c2c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,10 @@ macro(spirv_cross_add_library name config_name)
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
       $<INSTALL_INTERFACE:include/spirv_cross>)
   set_target_properties(${name} PROPERTIES
+      OUTPUT_NAME "${name}"
+      OUTPUT_NAME_DEBUG "${name}-d"
+      PDB_NAME "${name}"
+      PDB_NAME_DEBUG "${name}-d"
       PUBLIC_HEADERS "${hdrs}")
   target_compile_options(${name} PRIVATE ${spirv-compiler-options})
   target_compile_definitions(${name} PRIVATE ${spirv-compiler-defines})
